rero_ils/modules/ill_requests/forms.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Forms definitions about ILL request in public view."""
+
+from datetime import datetime
+
+from flask_babelex import gettext as _
+from flask_wtf import FlaskForm
+from wtforms import FormField, IntegerField, RadioField, SelectField, \
+    StringField, TextAreaField, validators
+from wtforms.fields.html5 import URLField
+
+from rero_ils.modules.utils import get_ref_for_pid
+from rero_ils.utils import remove_empties_from_dict
+
+
+class ILLRequestDocumentSource(FlaskForm):
+    """Form for 'published in' information about an ILL Request."""
+
+    class Meta:
+        """Meta class for ILLRequestDocumentSource."""
+
+        csrf = False
+
+    journal_title = StringField(
+        validators=[
+            validators.Optional(),
+            validators.Length(min=3)
+        ],
+        description=_('Journal or book title')
+    )
+    volume = StringField(
+        description=_('Volume'),
+        render_kw={'placeholder': '1, 2, ...'}
+    )
+    number = StringField(
+        description=_('Number'),
+        render_kw={'placeholder': '1, January, ...'}
+    )
+
+
+class ILLRequestDocumentForm(FlaskForm):
+    """Form for document about an ILL Request."""
+
+    class Meta:
+        """Meta class for ILLRequestDocumentForm."""
+
+        csrf = False
```

Comment:
why?

Reply:
As this form is part of a parent form, if I didn't disable the CSRF check on children forms, the form validation failed.
The main form (`ILLRequestForm`) has a CSRF validation enabled and working (tested)